packages/flutter/lib/src/rendering/sliver.dart

```diff
@@ -1442,6 +1442,23 @@ abstract class RenderSliver extends RenderObject {
     return null;
   }
 
+  /// This returns a absolute [Size] of the sliver.
+  ///
+  /// The dimensions are always positive and is only valid after [layout] has
+  /// completed.
+  ///
+  /// See also:
+  ///
+  ///  * [getAbsoluteSizeRelativeToOrigin], which returns relative size to the
+  ///    the leading edge of the sliver.
+  @protected
+  Size getAbsoluteSize() {
+    assert(geometry != null);
+    assert(!debugNeedsLayout);
+    final Size relative = getAbsoluteSizeRelativeToOrigin();
+    return Size(relative.width.abs(), relative.height.abs());
```

Comment:
Instead of calling abs on the size from getAbsoluteSizeRelativeToOrigin, can we just calculate the size directly from constraints.crossAxisExtent and geometry.paintExtent here?